Mathlib/Geometry/Manifold/Orientable.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rida Hamadani
+-/
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Orientable Manifolds
+
+This module defines orientable manifolds.
+
+## Main Definitions
+
+- `OrientationPreserving` : a map is orientation-preserving on a given set if the determinant of
+  its Jacobian is strictly positive on that set.
+- `orientationPreservingGroupoid` : the groupoid of partial homeos of `H` which are
+  orientation-preserving.
+- `OrientableManifold`: a type class saying that the charted space `M`, modelled on the space `H`,
+  admis an orientation.
+
+## TODO
+
+- Define orientable `0`-manifolds.
+
```

Comment:
Thinking about it more, it would be nice to mention the choice of definition also. What do you think about this (rewrapping lines as needed)?
```suggestion
## Implementation notes

The current definitions work for differentiable manifolds (and is simpler to define this way).
For topological manifolds, orientability can be defined using *local* orientations
(which mathlib cannot even state yet, as there is no e.g. singular homology).
In the future, it would be nice to generalise these definitions to allow for topological manifolds also, and relate them to the current definition.